src/move_basic.cpp

```diff
@@ -621,13 +639,14 @@ bool MoveBasic::moveLinear(double requestedDistance)
             velocity = 0;
         }
 
-        //ROS_INFO("%f %f %f", distTravelled, distRemaining, velocity);
-
-        if (distTravelled > requestedDistance - linearTolerance) {
+        if (std::abs(distTravelled) > std::abs(requestedDistance) - linearTolerance) {
             velocity = 0;
             done = true;
             ROS_INFO("Done linear, error %f meters", distRemaining);
         }
+        if (requestedDistance < 0) { 
+            velocity = -velocity;
```

Comment:
Does it make sense to have a separate velocity limit when going backward?

Reply:
I don't think so.